pkg/kubelet/volume_manager.go

```diff
@@ -39,10 +40,22 @@ func newVolumeManager() *volumeManager {
 
 // SetVolumes sets the volume map for a pod.
 // TODO(yifan): Currently we assume the volume is already mounted, so we only do a book keeping here.
-func (vm *volumeManager) SetVolumes(podUID types.UID, podVolumes kubecontainer.VolumeMap) {
+func (vm *volumeManager) SetVolumes(pod *api.Pod, podVolumes kubecontainer.VolumeMap) {
 	vm.lock.Lock()
 	defer vm.lock.Unlock()
+	podUID := pod.UID
 	vm.volumeMaps[podUID] = podVolumes
+
+	for _, volume := range podVolumes {
+		if volume.Builder.GetAttributes().VolumeGID != nil {
+			if pod.Spec.SecurityContext == nil {
+				pod.Spec.SecurityContext = &api.PodSecurityContext{}
+			}
```

Comment:
Go and Kube preference is to un-indent blocks like this.  Return or continue early on errors.

```
for ...
   if volume.Build.GetAttribtutes().VolumeGID == nil {
      continue 
  }
 // the rest ...
}
```